tests/test_implements_satisfy_protocols.py

```diff
@@ -1,45 +1,56 @@
 from typing import Protocol, runtime_checkable
 
-from dbt_semantic_interfaces.implementations.elements.dimension import (
-    PydanticDimension,
-    PydanticDimensionTypeParams,
-    PydanticDimensionValidityParams,
-)
+from hypothesis import given
+from hypothesis.strategies import builds, just, lists
+
+from dbt_semantic_interfaces.implementations.elements.dimension import PydanticDimension
 from dbt_semantic_interfaces.implementations.elements.entity import PydanticEntity
 from dbt_semantic_interfaces.implementations.elements.measure import PydanticMeasure
-from dbt_semantic_interfaces.implementations.metadata import (
-    PydanticFileSlice,
-    PydanticMetadata,
-)
+from dbt_semantic_interfaces.implementations.metadata import PydanticMetadata
 from dbt_semantic_interfaces.implementations.metric import (
     PydanticMetric,
+    PydanticMetricInput,
     PydanticMetricInputMeasure,
     PydanticMetricTypeParams,
 )
+from dbt_semantic_interfaces.implementations.project_configuration import (
+    PydanticProjectConfiguration,
+)
+from dbt_semantic_interfaces.implementations.saved_query import PydanticSavedQuery
 from dbt_semantic_interfaces.implementations.semantic_manifest import (
     PydanticSemanticManifest,
 )
-from dbt_semantic_interfaces.implementations.semantic_model import (
-    NodeRelation,
-    PydanticSemanticModel,
+from dbt_semantic_interfaces.implementations.semantic_model import PydanticSemanticModel
+from dbt_semantic_interfaces.implementations.time_spine_table_configuration import (
+    PydanticTimeSpineTableConfiguration,
 )
 from dbt_semantic_interfaces.protocols import Dimension as DimensionProtocol
 from dbt_semantic_interfaces.protocols import Entity as EntityProtocol
 from dbt_semantic_interfaces.protocols import Measure as MeasureProtocol
 from dbt_semantic_interfaces.protocols import Metadata as MetadataProtocol
 from dbt_semantic_interfaces.protocols import Metric as MetricProtocol
+from dbt_semantic_interfaces.protocols import SavedQuery as SavedQueryProtocol
 from dbt_semantic_interfaces.protocols import (
     SemanticManifest as SemanticManifestProtocol,
 )
 from dbt_semantic_interfaces.protocols import SemanticModel as SemanticModelProtocol
-from dbt_semantic_interfaces.type_enums import (
-    AggregationType,
-    DimensionType,
-    EntityType,
-    MetricType,
-    TimeGranularity,
+from dbt_semantic_interfaces.protocols.time_spine_configuration import (
+    TimeSpineTableConfiguration as TimeSpineTableConfigurationProtocol,
+)
+from dbt_semantic_interfaces.type_enums import MetricType
+
+SIMPLE_METRIC_STRATEGY = builds(
+    PydanticMetric,
+    type=just(MetricType.SIMPLE),
+    type_params=builds(PydanticMetricTypeParams, measure=builds(PydanticMetricInputMeasure)),
+)
+
+SEMANTIC_MODEL_STRATEGY = builds(
+    PydanticSemanticModel,
+    dimensions=lists(builds(PydanticDimension)),
+    entities=lists(builds(PydanticEntity)),
+    measures=lists(builds(PydanticMeasure)),
```

Comment:
Oh neat, it infers values based on type hints, I guess recursively all the way down.

Reply:
It does it pretty well! Although it sometimes jokes on nested lists of structured objects and just does an empty list instead. Hence why I explicitly tell it to generate lists of dimensions/entities/measures

Reply:
It also might be choking on nested objects in general. Or maybe it's only nested optional objects 🤔